src/js/components/TextInput/TextInput.js

```diff
@@ -99,7 +100,12 @@ const TextInput = forwardRef(
     const suggestionsRef = useRef();
     const suggestionRefs = {};
 
-    const [value, setValue] = formContext.useFormContext(name, valueProp);
+    // if this is a readOnly property, don't set a name with the form context
+    // this allows Select to control the form context for the name.
+    const [value, setValue] = formContext.useFormContext(
+      readOnly ? undefined : name,
+      valueProp,
```

Comment:
Reading through those comments, I don't see how Select is using the readOnly prop to control formContext for the name, what am I missing?

Reply:
That was existing already: https://github.com/grommet/grommet/blob/546e18c9fea9d4e888d6437cf39b1a3453f1eb0c/src/js/components/Select/Select.js#L246